.github/workflows/type-check.yml

```
@@ -17,9 +17,8 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-python@v4
-      - run: pip install ".[jupyter-executor]" mypy
-      # As more modules are type check clean, add them here
+      # All additional modules should be defined in setup.py
+      - run: pip install ".[types]"
```

Rather than merging `jupyter-executor` into `types` I think we should install both here
```suggestion
      - run: pip install ".[types, jupyter-executor]"
```

---

The main idea is that all dependancies must be declared in setup.py, otherwise it is difficult for other developers to figure out how to update them. Workflow files should also be fixed and not be changed every time we update something in the project settings.

---

Then shouldn't types install all extras and not just jupyter-executor?